front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts

```diff
@@ -79,8 +81,16 @@ const useLazyProjectTrains = ({
       packageToProject.forEach((trainId) => requestedProjectedTrainIds.current.add(trainId));
 
       // Format train ids back to editoast format
-      const editoastTrainIds = packageToProject.map((trainId) =>
-        formatTrainScheduleIdToEditoastTrainId(trainId as TrainScheduleId)
+      const { editoastTrainIds, editoastPacedTrainIds } = packageToProject.reduce(
```

Comment:
nit: This should work to avoid the cast
```suggestion
      const { editoastTrainIds, editoastPacedTrainIds } = packageToProject.reduce<{ editoastTrainIds: number[], editoastPacedTrainIds: number[] }>(
        (acc, trainId) => {
          if (isTrainSchedule(trainId)) {
            acc.editoastTrainIds.push(formatTrainScheduleIdToEditoastTrainId(trainId));
          } else {
            acc.editoastPacedTrainIds.push(formatPacedTrainIdToEditoastTrainId(trainId));
          }
          return acc;
        },
        { editoastTrainIds: [], editoastPacedTrainIds: [] }
```

Reply:
done